src/rcvlink.rs

```diff
@@ -361,6 +363,17 @@ impl ReceiverLinkBuilder {
         self
     }
 
+    /// Set or reset a receive link property
+    pub fn property(mut self, name: &str, value: Option<&str>) -> Self {
```

Comment:
Is it correct to accept only strings? I have set props such as 

            .property("com.microsoft:entity-type", Some("8"))
            .property(
                "apache.org:selector-filter:string",
                Some("amqp.annotation.x-opt-offset>'@latest'"),
            )
            
and I can see in frame trace these even come back with the response, but it doesn't have any effect on filtering.

Reply:
Used Symbol and Variant for property parameters:

            .property(
                Symbol::from_slice("com.microsoft:entity-type"),
                Some("8".into()),
            )
            .property(
                Symbol::from_slice("apache.org:selector-filter:string"),
                Some(Variant::Described((
                    Descriptor::Symbol(Symbol::from_slice("apache.org:selector-filter:string")),
                    Box::new(Variant::from("amqp.annotation.x-opt-offset > '@latest'")),
                ))),

Still no effect, though the response seems to indicate the server understood:

    AmqpFrame {
        channel_id: 0,
        performative: Attach(
            Attach {
                name: "mylink31",
                handle: 31,
                role: Sender,
                snd_settle_mode: Settled,
                rcv_settle_mode: First,
                source: Some(
                    Source {
                        address: Some(
                            "device-telemetry/ConsumerGroups/ehaus/Partitions/31",
                        ),
                        durable: None,
                        expiry_policy: SessionEnd,
                        timeout: 0,
                        dynamic: false,
                        dynamic_node_properties: None,
                        distribution_mode: None,
                        filter: None,
                        default_outcome: None,
                        outcomes: None,
                        capabilities: None,
                    },
                ),
                target: None,
                unsettled: None,
                incomplete_unsettled: false,
                initial_delivery_count: Some(
                    0,
                ),
                max_message_size: Some(
                    65535,
                ),
                offered_capabilities: None,
                desired_capabilities: None,
                properties: Some(
                    {
                        Symbol(
                            B:"client-id",
                        ): String(
                            B:"0d0ef7654d454c659021cf177c69cdbe",
                        ),
                        Symbol(
                            B:"apache.org:selector-filter:string",
                        ): Described(
                            (
                                Ulong(
                                    83483426826,
                                ),
                                String(
                                    B:"amqp.annotation.x-opt-offset > '@latest'",
                                ),
                            ),
                        ),
                        Symbol(
                            B:"com.microsoft:entity-type",
                        ): String(
                            B:"8",
                        ),
                        Symbol(
                            B:"com.microsoft:timeout",
                        ): Uint(
                            64980,
                        ),
                    },
                ),
            },
        ),
    }